frontend/src/pages/connectionTypes/manage/advanced/DropdownAdvancedPropertiesForm.tsx

```diff
@@ -0,0 +1,208 @@
+import * as React from 'react';
+import { Button, Flex, FlexItem, FormGroup, Radio, TextInput } from '@patternfly/react-core';
+import { MinusCircleIcon, PlusCircleIcon } from '@patternfly/react-icons';
+import { Table, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import text from '@patternfly/react-styles/css/utilities/Text/text';
+import useDraggableTableControlled from '~/utilities/useDraggableTableControlled';
+import { DropdownField } from '~/concepts/connectionTypes/types';
+import { AdvancedFieldProps } from '~/pages/connectionTypes/manage/advanced/types';
+import ExpandableFormSection from '~/components/ExpandableFormSection';
+
+const DropdownAdvancedPropertiesForm: React.FC<AdvancedFieldProps<DropdownField>> = ({
+  properties,
+  onChange,
+  onValidate,
+}) => {
+  React.useEffect(() => {
+    if (!properties.variant) {
+      onChange({ ...properties, variant: 'single' });
+    }
+  });
+
+  React.useEffect(() => {
+    // TODO: warn user of duplicate values in table
+    const duplicates = properties.items?.find((item1, index1) =>
+      properties.items?.find((item2, index2) => item1.value === item2.value && index1 !== index2),
+    );
+    onValidate(
```

Comment:
Jeff's latest commit allows for empty fields to be provided in file extensions but then does a clean up of empty rows on submit. I think that gives more flexibility for the user and can likely be used here as well. He added a single location to perform clean up where you can duplicate what is done.